File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.py
from collections import deque

def solution(queue1, queue2):
    answer = 0
    que1 = deque(queue1)
    que2 = deque(queue2)
    
    sum1 = sum(que1)
    sum2 = sum(que2)
    
    while True:
        if answer > len(queue1)*4:
            return -1
        if sum1<sum2:
            val = que2.popleft()
            que1.append(val)
            sum1+=val
            sum2-=val
        elif sum1>sum2:
            val = que1.popleft()
            que2.append(val)
            sum1-=val
            sum2+=val
        else:
            return answer
        answer+=1
    return -1

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.java
import java.util.*;

class Solution {
    public int solution(int[] queue1, int[] queue2) {
        long sum1=0;
        long sum2=0;
        int cnt=0;


        Queue<Integer> que1 = new LinkedList<>();
        Queue<Integer> que2 = new LinkedList<>();



        for(int q : queue1) {
            sum1 += q;
            que1.offer(q);
        }
        for(int q : queue2) {
            sum2+= q;
            que2.offer(q);
        }
        if((sum1+sum2)%2 == 1) return -1;
        
        
        long target = (sum1+sum2)/2;

        while(cnt <= queue1.length * 3){
            if(sum1 == target) return cnt;

            if(sum1>target){
                int tmp = que1.poll();
                sum1 -= tmp;
                que2.offer(tmp);
            }else{
                int tmp = que2.poll();
                sum1 += tmp;
                que1.offer(tmp);
            }
            cnt++;
        }
        return -1;
    }
}